### PR TITLE
Remove restrictions on ONNXPow

### PR DESCRIPTION
**Pull request by josel-amd**
This now lowers to TOSA but what you get is which I think it along the lines of other operators.
```
onnx-mlir-opt single.mlir --shape-inference  --convert-onnx-to-tosa
module {
  func.func @test_less_broadcast(%arg0: tensor<5xi32>, %arg1: tensor<5xi32>) -> tensor<5xi32> {
    %0 = "onnx.Pow"(%arg0, %arg1) : (tensor<5xi32>, tensor<5xi32>) -> tensor<5xi32>
    onnx.Return %0 : tensor<5xi32>
  }
}
```

### CODE REVIEW

**Review by mgehre-amd — APPROVED** [review]

File: src/Dialect/ONNX/ONNXOps/Math/ElementwiseBroadcast.cpp [hunk 1]
```diff
@@ -343,15 +343,7 @@ LogicalResult ONNXOrOp::inferShapes(
 //===----------------------------------------------------------------------===//
 
 LogicalResult ONNXPowOp::verify() {
-  ShapedType lhsTy = getX().getType().cast<ShapedType>();
-  ShapedType rhsTy = getY().getType().cast<ShapedType>();
-  Type rhsETy = rhsTy.getElementType();
-  Type lhsETy = lhsTy.getElementType();
-  if (rhsETy != lhsETy)
-    return emitOpError("Pow with different input type not implemented yet");
```
Comment (cferry-AMD) [inline src/Dialect/ONNX/ONNXOps/Math/ElementwiseBroadcast.cpp:351]: The new test only checks the shape, not the types. Is this actually supported?

Comment (josel-amd) [reply src/Dialect/ONNX/ONNXOps/Math/ElementwiseBroadcast.cpp:351]: I think it is important to define "supported" here. I defined "supported" as being able to represent an operator in onnx-mlir, by adhering to the dialect specification. In that spirit, the restrictions that were here before do not apply. Now if you ask if this is supported in TOSA, the answer is no but, as it happens with other operators, the `--convert-onnx-to-tosa` should not fail but keep the (unmodified) operators that it failed to convert which is what I'm doing now.

Comment (cferry-AMD) [reply src/Dialect/ONNX/ONNXOps/Math/ElementwiseBroadcast.cpp:351]: Discussed offline, looks fine. Thanks!

**Review by cferry-AMD — APPROVED** [review]

File: test/mlir/conversion/onnx_to_tosa/Math/Pow.mlir [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// RUN: onnx-mlir-opt --shape-inference --convert-onnx-to-tosa %s -split-input-file | FileCheck %s
```
Comment (TinaAMD) [inline test/mlir/conversion/onnx_to_tosa/Math/Pow.mlir:1]: Maybe name the file `Pow-unsupported` such that it is clear this is only cases which cannot be lowered?

**Review by TinaAMD — APPROVED** [review]
LGTM

File: test/mlir/conversion/onnx_to_tosa/Math/Pow.mlir (hunk 2 above)
Comment (josel-amd) [reply test/mlir/conversion/onnx_to_tosa/Math/Pow.mlir]: Done!

**Pull request merged**